### PR TITLE
QAR- 101 - AR screen to location

### PR DESCRIPTION
**Pull request by LDuncAndroid**
https://devtopia.esri.com/runtime/queen-annes-revenge/issues/101

- Added function to enable screen to location
- Updated ArcGIS lib to 100.6.0

### CODE REVIEW

**Comment by LDuncAndroid** [comment]
Assigning to @mhdostal for first review. Thanks.

**Comment by mhdostal** [comment]
@LDuncAndroid - had a couple of other comments:

- the `translationTransformationFactor` [here](https://github.com/Esri/arcgis-runtime-toolkit-android/blob/c22a329ba52b01c9a65cb7ebb43a27df977fa1f4/arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt#L381) is now called `translationFactor`.

- the `originCamera` and `translationTransformationFactor` properties appear to set the corresponding values in the `cameraController`.  However, at least the `translationFactor` can be set outside the Toolkit code (pinch to zoom), so I'm wondering if the ArcGISARView property would be out of date then.

On iOS, I don't store the translationFactor but get/set it from the CameraController.

**Comment by LDuncAndroid** [comment]
Thanks for the review @mhdostal . I've made those changes.

**Comment by LDuncAndroid** [comment]
@mhdostal I had to make another commit after requesting review, sorry. Missed the case where trying to start ARCore when it wasn't installed.

**Comment by mhdostal** [comment]
> * the `originCamera` and `translationTransformationFactor` properties appear to set the corresponding values in the `cameraController`.  However, at least the `translationFactor` can be set outside the Toolkit code (pinch to zoom), so I'm wondering if the ArcGISARView property would be out of date then.

The only other thing I think you should test is the above.  Both `translationFactor` and `originCamera` can change outside the toolkit from user interaction on the scene.  I'd check and make sure that those values are updated when that happens.

**Comment by LDuncAndroid** [comment]
> > * the `originCamera` and `translationTransformationFactor` properties appear to set the corresponding values in the `cameraController`.  However, at least the `translationFactor` can be set outside the Toolkit code (pinch to zoom), so I'm wondering if the ArcGISARView property would be out of date then.
> 
> The only other thing I think you should test is the above. Both `translationFactor` and `originCamera` can change outside the toolkit from user interaction on the scene. I'd check and make sure that those values are updated when that happens.

@mhdostal they won't be automatically updated when the change happens in `SceneView` but and call to the getter/setter will get/set from the camera controller. I have missed the `originCamera` as you've described above.

But in the issue with the Behavioural changes, you mentioned setting the `originCamera` to `nil` in `resetTracking()`, but doing so will cause an exception.

**Comment by mhdostal** [comment]
@LDuncAndroid - Approved.

**Comment by mhdostal** [comment]
> But in the issue with the Behavioural changes, you mentioned setting the `originCamera` to `nil` in `resetTracking()`, but doing so will cause an exception.

I'm going to revisit this issue on iOS when the final design for the StartTracking enum is approved.

**Comment by LDuncAndroid** [comment]
Thanks for the review @mhdostal . Assigning to @gunt0001 for 2nd review.

File: arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt [hunk 1]
```diff
@@ -655,6 +661,31 @@ class ArcGISArView : FrameLayout, DefaultLifecycleObserver, Scene.OnUpdateListen
         return false
     }
 
+    /**
+     * Performs a hitTest at the [screenPoint]. If the hitTest succeeds this returns the point in the
+     * world where the hitTest happened
```
Comment (gunt0001) [inline arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt]: instead of "world" better use the term "scene"?

Comment (gunt0001) [reply arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt]: Missing full stop .

File: toolkit-test-app/src/main/java/com/esri/arcgisruntime/toolkit/test/ar/ArcGISArViewActivity.kt [hunk 2]
```diff
@@ -303,11 +312,28 @@ class ArcGISArViewActivity : AppCompatActivity() {
...
+                        if (currentScene?.isTabletop == true) {
                             arcGisArView.setInitialTransformationMatrix(this)
+                        } else {
+                            var sphere = SimpleMarkerSceneSymbol.createSphere(
+                                Color.CYAN,
+                                0.25,
+                                SceneSymbol.AnchorPosition.BOTTOM
+                            )
+                            arcGisArView.arScreenToLocation(this)?.let {
+                                if (sphereOverlay == null) {
```
Comment (gunt0001) [inline toolkit-test-app/src/main/java/com/esri/arcgisruntime/toolkit/test/ar/ArcGISArViewActivity.kt]: Could you make `sphereOverlay` a lazy property and have all of this logic of creating/adding the GraphicsOverlay in the property implementation?

Comment (LDuncAndroid) [reply toolkit-test-app/src/main/java/com/esri/arcgisruntime/toolkit/test/ar/ArcGISArViewActivity.kt]: 🤦‍♂ yeah. I don't know why I did it this way.

File: arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt (hunk 1 above)
Comment (LDuncAndroid) [reply arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt]: I've changed it to 'AR scene' to differentiate between the ArcGIS scene and AR scene.

**Comment by LDuncAndroid** [comment]
Thanks for the review @gunt0001 . I've fixed those issues. Ready for re-review.

**Comment by LDuncAndroid** [comment]
Thanks for the reviews @mhdostal , @gunt0001 . Merging.

**Pull request merged**